From 8a5cdacfa4d0adf4af2f98713ff03f275be19e26 Mon Sep 17 00:00:00 2001
From: Gao Pan <pandy.gao@nxp.com>
Date: Fri, 1 Dec 2017 13:17:07 +0800
Subject: [PATCH 2953/5242] MLK-17033: ath10k: fix suspend/resume fail issue

commit  c857313799e3cb8ffe46d59cc421d363c0a46afc from
https://source.codeaurora.org/external/imx/linux-imx.git

qca6174 wifi driver causes system hang during suspend/resume stress test.
This patch fix this suspend/resume fail issue.

Signed-off-by: Andy Duan <fugang.duan@nxp.com>
Signed-off-by: Gao Pan <pandy.gao@nxp.com>
(cherry-picked from 72fbaf4aa36e7407108d1c0b7d857287f84bee3d)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/wireless/ath/ath10k/pci.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/wireless/ath/ath10k/pci.c b/drivers/net/wireless/ath/ath10k/pci.c
index af2cf55..a062315 100644
--- a/drivers/net/wireless/ath/ath10k/pci.c
+++ b/drivers/net/wireless/ath/ath10k/pci.c
@@ -2069,6 +2069,8 @@ static void ath10k_pci_hif_stop(struct ath10k *ar)
 	ath10k_pci_irq_disable(ar);
 	ath10k_pci_irq_sync(ar);
 	ath10k_pci_flush(ar);
+	ath10k_pci_sleep_sync(ar);
+
 	napi_synchronize(&ar->napi);
 	napi_disable(&ar->napi);
 
-- 
1.7.9.5

